### ISSUE TITLE
Issues Building Demo Application

### ISSUE THREAD

**Issue by jibright** [aws/amazon-chime-sdk-cpp#24]
Several Issues present when building demo application using steps from: https://github.com/aws/amazon-chime-sdk-cpp/blob/main/chime-sdk-signaling-cpp/demo/cli/README.md

Step 2. branch 4844 includes use of libFuzzer which fails to download. Had to use a newer branch that is no longer dependant on libFuzzer (>6000)

Step 6. Issue with type conversion
`error: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
  258 |                             CONF_VALUE *nval = OPENSSL_sk_value((const OPENSSL_STACK *)&cv, j);`
Changed 
`int j;` to 
`size_t j;` on line 251 of file chime-sdk-signaling-cpp/build/_deps/libwebsockets-src/lib/tls/openssl/openssl-x509.c
and that fixed this problem, but I shouldn't need to be changing code.

Step 7. Issue with protubuf version issue between chime sdk and webrtc. Added
 `target_include_directories(${CLI_NAME} PUBLIC <home_folder>/amazon-chime-sdk-cpp-main/chime-sdk-signaling-cpp/build/_deps/protobuf-src/src/)`
to end of demo/cli CmakeLists.txt to force it to use the correct protobuf version.

`cmake -S . -B build -DCMAKE_TOOLCHAIN_FILE="${TOOLCHAIN_FILE}"`  runs successfully, but when I run `cmake --build build` I get the following error:
`webrtc-build/webrtc/api/peer_connection_interface.h:1569:16: error: no viable conversion from 'rtc::scoped_refptr<VideoTrackSourceInterface>' (aka 'scoped_refptr<webrtc::VideoTrackSourceInterface>') to 'VideoTrackSourceInterface *'
 1569 |         label, rtc::scoped_refptr<VideoTrackSourceInterface>(source));`

I am not sure where to go from here. I believe the issues stem from the guide being quite old, and using an older version of WebRTC, seeing how it refers to using branch 4844 which is 2 years old. Please advise as we would like to move forward with this project using the chime service.

Thanks for your help.

**Comment by comoc** [comment]
I am in the same situation as @jibright .

Branch `4844` does not contain the `third_party` directory.
https://chromium.googlesource.com/external/webrtc/+/branch-heads/4844
Therefore, `gclient sync` fails.

Any information would be appreciated.

Thank you.

**Comment by jbauers** [comment]
Hi there! Echoing @jibright, we're also experiencing issues, starting with `libFuzzer`:

```
#18 499.2 33>WARNING: subprocess '"git" "-c" "core.deltaBaseCacheLimit=2g" "fetch" "origin" "debe7d2d1982e540fbd6bd78604bf001753f9e74" "--no-tags"' in /home/build/src/third_party/libFuzzer/src failed; will retry after a short nap...
#18 502.0 33>WARNING: subprocess '"git" "-c" "core.deltaBaseCacheLimit=2g" "fetch" "origin" "debe7d2d1982e540fbd6bd78604bf001753f9e74" "--no-tags"' in /home/build/src/third_party/libFuzzer/src failed; will retry after a short nap...
#18 505.2 
#18 505.2 src/third_party/libFuzzer/src (ERROR)
#18 505.2 ----------------------------------------
#18 505.2 [0:00:38] Started.
#18 505.2 [0:00:38] Finished running: git rev-list -n 1 HEAD
#18 505.2 [0:00:38] Finished running: git rev-parse --abbrev-ref=strict HEAD
#18 505.2 error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400 Bad Request
#18 505.2 [0:00:39] fatal: The remote end hung up unexpectedly
#18 505.2 [0:00:41] error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400 Bad Request
#18 505.2 [0:00:41] fatal: The remote end hung up unexpectedly
#18 505.2 ----------------------------------------
#18 505.2 Error: Command 'git -c core.deltaBaseCacheLimit=2g fetch origin debe7d2d1982e540fbd6bd78604bf001753f9e74 --no-tags' returned non-zero exit status 128 in /home/build/src/third_party/libFuzzer/src
#18 ERROR: process "/bin/sh -c fetch --nohooks webrtc  && cd src  && git checkout -b _my__m99_branch refs/remotes/branch-heads/4844  && cd ..  && gclient sync  && gclient sync -D" did not complete successfully: exit code: 1
```

Using a newer branch then leads to more issues during the build process.

Is there any plan to update this repository / the documentation to make this buildable again? Thanks!

**Comment by hensmi-amazon** [comment]
Hi sorry for the delay, can you retry with my latest changes and README update? I have updated the build to use the latest libwebrtc.

**Comment by georgezy-amzn** [comment]
This issue has been fixed with PR #29 on Aug 9. Please reopen or submit a support case if this is still an issue.
